### ISSUE TITLE
persist_docs should consider column quoting

### ISSUE THREAD

**Issue by drewbanin** [fishtown-analytics/dbt#2539]
### Describe the bug
When running a `dbt run`, sql statements to persist docs are failing for columns that must be quoted. dbt does not currently respect the [quote](https://docs.getdbt.com/reference/resource-properties/quote) config when building these DDL statements.

### Steps To Reproduce
```
# dbt_project.yml
name: debug
version: 1.0.0
config-version: 2

models:
  +persist_docs:
    columns: true
```

```
-- models/my_model.sql
select 1 as "2id"
```

```
# models/schema.yml

version: 2

models:
  - name: debug
    columns:
      - name: 2id
        description: "my description"
        quote: true
```

Finally, execute `dbt run` and observe that persisting docs fails.

### Expected behavior
The DDL built by dbt should respect the `quote` config on the `column` declaration for a dbt node

### System information
**Which database are you using dbt with?**
- [x] postgres
- [x] redshift
- [ ] bigquery
- [x] snowflake

This probably does not affect BigQuery since quoting isn't relevant + we use the API

**The output of `dbt --version`:**
```
0.17.0
```

**Comment by drewbanin** [comment]
This is not a regression per se - moving it to the Marian Anderson milestone

**Comment by jtcohen6** [comment]
Resolved by #2733
